iot-hub/Samples/device/FileUploadSample/FileUploadSample.cs

```diff
@@ -48,8 +49,13 @@ public async Task RunSampleAsync()
 
                 // Note that other versions of the Azure Storage SDK can be used here. For the latest version, see
                 // https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage#azure-storage-libraries-for-net
-                var blobClient = new BlobClient(uploadUri);
-                await blobClient.UploadAsync(fileStreamSource, overwrite: true);
+                // NOTE: The UploadAsync operation overwrites the contents of the blob, creating a new block blob if none exists.
+                // Overwriting an existing block blob replaces any existing metadata on the blob.
+                // Set <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-conditional-headers-for-blob-service-operations">
+                // access conditions through BlobRequestConditions to avoid overwriting existing data.
+                // See https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs#L473 for more details.

```

Comment:
Not sure what the answer for this question is, but isn't referencing line numbers in code problematic when BlockBlobClient.cs changes? Because the content on line 473 is no longer relevant.

Reply:
good point! this should have been a permalink, I'll update it